contrib/test.sh

```diff
@@ -8,41 +10,50 @@ then
     alias cargo="cargo +$TOOLCHAIN"
 fi
 
+cargo --version
+rustc --version
+
 # Format if told to
 if [ "$DO_FMT" = true ]
 then
-    (
-        rustup component add rustfmt
-        cargo fmt --all -- --check
-    )
+    rustup component add rustfmt
+    cargo fmt --all -- --check
 fi
 
 # Fuzz if told to
 if [ "$DO_FUZZ" = true ]
 then
-    (
-        cd fuzz
-        cargo test --verbose
-        ./travis-fuzz.sh
-        # Exit out of the fuzzer,
-        # run stable tests in other CI vms
-        exit 0
-    )
+    cd fuzz
+    cargo test --verbose
+    ./travis-fuzz.sh
+
+    # Exit out of the fuzzer, do not run other tests.
+    exit 0
 fi
 
-# Test without any features first
-cargo test --verbose
+# Defaults / sanity checks
+cargo test
 
-# Test each feature
-for feature in ${FEATURES}
-do
-    cargo test --verbose --features="$feature"
-done
+if [ "$DO_FEATURE_MATRIX" = true ]
+then
+    # All features
+    cargo test --features="$FEATURES"
+
+    # Single features
+    for feature in ${FEATURES}
+    do
+        cargo test --features="$feature"
+    done
 
-# Also build and run each example to catch regressions
-cargo build --examples
-# run all examples
-run-parts ./target/debug/examples
+    # Run all the examples
+    cargo build --examples
+    cargo run --example htlc --features=compiler 
```

Comment:
nit: There is a white space at the end of this line.

Reply:
ooo, means I did not read the diff before pushing, bad Tobin. Thanks man, I'll fix it.

Reply:
I read the diff and I missed this too, oops! Thanks Sanket.